### DISCUSSION TITLE
Ordering tasks

### DISCUSSION THREAD

**Discussion by adamalexandru4** [kagkarlsson/db-scheduler#572] [General]
I’m wondering if it’s possible to organize tasks similarly to how Kafka handles messages within topic partitions.

For example, when scheduling tasks like task1, task2, and task3, you could provide additional information (e.g., a key). The scheduler would then process the tasks sequentially: first completing task1, then task2, and finally task3, ensuring they are never executed in parallel or out of order.

Does anyone have any ideas?

**Comment by kagkarlsson** [comment]
The only thing you can do is to have one task schedule the next (or "change into" the next)

[JobChainingUsingTaskDataMain.java](https://github.com/kagkarlsson/db-scheduler/blob/master/examples/features/src/main/java/com/github/kagkarlsson/examples/JobChainingUsingTaskDataMain.java)

[JobChainingUsingSeparateTasksMain.java](https://github.com/kagkarlsson/db-scheduler/blob/master/examples/features/src/main/java/com/github/kagkarlsson/examples/JobChainingUsingSeparateTasksMain.java)

**Comment by adamalexandru4** [comment]
The problem is that jobs can come later than I schedule the first task. I don't know from the first place which tasks to run. They come from external services but I need to execute them in the receiving order.

**Comment by ericsouza** [comment]
I miss this feature too. Python has a cool library called [procrastinate](https://procrastinate.readthedocs.io/en/stable/) where we can set a lock key for each dispatched task which will be used to achieve a [ordering mechanism](https://procrastinate.readthedocs.io/en/stable/howto/advanced/locks.html)
